docs/fastdds/xtypes/discovery_matching.rst

```diff
@@ -0,0 +1,102 @@
+.. include:: ../../03-exports/aliases-api.include
+.. include:: ../../03-exports/aliases.include
+
+.. _xtypes_discovery_matching:
+
+Discovery and Endpoint Matching
+===============================
+
+.. _DDS-XTypes V1.3: https://www.omg.org/spec/DDS-XTypes/1.3
+
+*eProsima Fast DDS* incorporates an advanced type discovery service used on endpoint discovery and matching.
+Discovery process leverages the optional |TypeInformation-api| value to ensure compatibility between communicating endpoints.
+Upon the discovery of a remote endpoint, *eProsima Fast DDS* examines the |TypeInformation-api| to determine
+the availability of essential data type information.
+
+The process of checking the types is as follows:
+
+1. **Evaluation of TypeInformation**: If the discovered enpoint has transmitted |TypeInformation-api|,
+     *eProsima Fast DDS* evaluates whether all required types are already known.
+     In cases where some types are missing, an automatic process to discover the necessary types is initiated.
+     |TypeObject-api| is used to communicate with other participants and to discover the topic types asynchronously.
+     Matching of the discovered enpoint is deferred until all necessary types are known.
+
+2. **Fallback Mechanism**: In scenarios where the discovered endpoint has not transmitted |TypeInformation-api|,
+     *eProsima Fast DDS* resorts to a fallback mechanism.
+     This mechanism relys on the type name to verify if the type is already known within the system.
+     If the type name is recognized, matching occurs.
```

Comment:
I suggest rewriting this section. I think there are some implementation details that are not interesting from the point of view of the user. I will focus the text on what the user needs to know and avoid any internal implementation detail.

Reply:
Done in 1f40ec6245de6255b8f3189ef10ca2fdf1c17fea